### PR TITLE
refactor: store value changes instead

### LINKED ISSUE #860: Proper model for PropertyValue values
### Describe the chore

Currently, we cast it into an interface{}, which throws away vital information.

Is this a string because it's an option, or a text, or a multi-select option?

We could embrace TypedValue and give it proper serialization to esdb

### PR DESCRIPTION
**Pull request by FoseFx**
## Which issues does this pull request close?
Cherry-picked from the now abandoned op. concurrency branch

Closes #860

### CODE REVIEW

File: services/property-svc/internal/property-value/aggregate/aggregate.go [hunk 1]
```diff
@@ -72,7 +75,12 @@ func (a *PropertyValueAggregate) onPropertyValueUpdated(evt hwes.Event) error {
 		return err
 	}
 
-	a.PropertyValue.Value = payload.Value
+	if payload.Change.ValueRemoved {
+		a.PropertyValue.Value = nil
+		return nil
+	}
+
+	payload.Change.Apply(a.PropertyValue.Value)
```
Comment (MaxSchaefer) [inline services/property-svc/internal/property-value/aggregate/aggregate.go:83]: We do not need to re-assign this due to the reference above?

File: services/property-svc/internal/property-value/projections/property_value_postgres_projection/property_value_postgres_projection.go [hunk 2]
```diff
@@ -91,38 +86,28 @@ func (p *Projection) onPropertyValueCreated(ctx context.Context, evt hwes.Event)
...
-			val, ok := payload.Value.(string)
-			if !ok {
-				return errors.New("could not assert string"), hwutil.PtrTo(esdb.NackActionPark)
-			}
-			id, err := uuid.Parse(val)
+		if isSingleSelectChange {
+			id, err := uuid.Parse(*change.SingleSelectValue)
 			if err != nil {
-				return err, hwutil.PtrTo(esdb.NackActionPark)
+				return fmt.Errorf("option change is not a uuid: %w", err), hwutil.PtrTo(esdb.NackActionSkip)
```
Comment (MaxSchaefer) [inline services/property-svc/internal/property-value/projections/property_value_postgres_projection/property_value_postgres_projection.go:104]: Why skip? We should park and investigate such issues.

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

File: services/property-svc/internal/property-value/aggregate/aggregate.go (hunk 1 above)
Comment (FoseFx) [reply services/property-svc/internal/property-value/aggregate/aggregate.go:83]: which reference above?

File: services/property-svc/internal/property-value/projections/property_value_postgres_projection/property_value_postgres_projection.go (hunk 2 above)
Comment (FoseFx) [reply services/property-svc/internal/property-value/projections/property_value_postgres_projection/property_value_postgres_projection.go:104]: the payload will never be valid, so parking makes little sense

File: services/property-svc/internal/property-value/aggregate/aggregate.go (hunk 1 above)
Comment (MaxSchaefer) [reply services/property-svc/internal/property-value/aggregate/aggregate.go:83]: Line #67

**Review by MaxSchaefer — APPROVED** [review]

File: services/property-svc/internal/property-value/aggregate/aggregate.go (hunk 1 above)
Comment (FoseFx) [reply services/property-svc/internal/property-value/aggregate/aggregate.go:83]: that's in onPropertyValueCreated, not onPropertyValueUpdated

**Pull request merged**